### PR TITLE
#1287 Fix drawing of all-day events.

### PR DESCRIPTION
**Pull request by RobertZenz**
`allDay` is the condition I was looking for, it allows to determine whether the event has a time at all.

Two points I'm unsure about:

 1. Did I place the comments correctly or would you prefer it *above* the `if` condition.
 2. Are there utility functions for determining whether it is the same day or not? Subtracting and checking the minutes works, but if you've got something better that would be it.

### CODE REVIEW

**Comment by RobertZenz** [comment]
On second thought...is `< MINUTES_PER_DAY` or `<= MINUTES_PER_DAY` correct? I'd wager `<`, right?

File: app/src/main/java/com/android/calendar/DayView.java [hunk 1]
```diff
@@ -2943,18 +2943,22 @@ private void drawAllDayEvents(int firstDay, int numDays, Canvas canvas, Paint p)
...
             }
-            int leftoffset;
-            int rightoffset;
+            int leftoffset = 0;
+            int rightoffset = 0;
             if (startDay < firstDay) {
                 startDay = firstDay;
-                leftoffset = 0;
-            } else {
+            } else if ((firstDay - startDay) < MINUTES_PER_DAY && !event.allDay) {
```
[the remark this answers is not in the document]
Comment (jspricke) [reply app/src/main/java/com/android/calendar/DayView.java:2950]: Maybe you don't even need the else and make it an extra if?

[the remark this answers is not in the document]
Comment (RobertZenz) [reply app/src/main/java/com/android/calendar/DayView.java:2950]: True, not sure what I tested there but I thought it was needed for correct drawing int eh day view. But works correctly, I've directly amended the changes.

[the remark this answers is not in the document]
Comment (jspricke) [reply app/src/main/java/com/android/calendar/DayView.java]: Can you try without the `else` as well?

[the remark this answers is not in the document]
Comment (RobertZenz) [reply app/src/main/java/com/android/calendar/DayView.java]: You mean like this?

```java
           if (startDay < firstDay) {
                startDay = firstDay;
            }
            if (!event.allDay) {
                // Only offset the drawing if it is not an all-day event (which
                // does not have a time at all).
                leftoffset = (event.startTime * cellWidth) / MINUTES_PER_DAY;
            }
            if (endDay > lastDay) {
                endDay = lastDay;
            }
            if (!event.allDay) {
                // Only offset the drawing it is not an all-day event (which
                // does not have a time at all).
                rightoffset =
                    ((MINUTES_PER_DAY - event.endTime) * cellWidth) / MINUTES_PER_DAY;
            }
```

Then every multi-day event is misdrawn in the day view, as the left and right offset are always calculated for the currently displayed day.

[the remark this answers is not in the document]
Comment (jspricke) [reply app/src/main/java/com/android/calendar/DayView.java]: I see, thanks for trying :).

**Review by jspricke — APPROVED** [review]

**Pull request merged**